src/test/java/com/rocketden/main/service/GameManagementServiceTests.java

```diff
@@ -174,6 +188,126 @@ public void startGameSuccess() {
         assertEquals(room.getDuration(), game.getGameTimer().getDuration());
     }
 
+    @Test
+    public void startGameWithProblemAndRandomProblemSuccess() {
+        User host = new User();
+        host.setNickname(NICKNAME);
+        host.setUserId(USER_ID);
+
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+        room.setHost(host);
+        room.setDifficulty(ProblemDifficulty.HARD);
+        room.setNumProblems(10);
+
+        Problem problem = new Problem();
+        problem.setProblemId(PROBLEM_ID);
+        problem.setName(PROBLEM_NAME);
+        problem.setDescription(PROBLEM_DESCRIPTION);
+        problem.setDifficulty(ProblemDifficulty.EASY);
+
+        Problem problem2 = new Problem();
+        problem.setProblemId(PROBLEM_ID_2);
+        problem.setName(PROBLEM_NAME_2);
+        problem.setDescription(PROBLEM_DESCRIPTION_2);
+        problem.setDifficulty(ProblemDifficulty.HARD);
+
+        room.setProblems(Collections.singletonList(problem));
+
+        StartGameRequest request = new StartGameRequest();
+        request.setInitiator(UserMapper.toDto(host));
+
+        Mockito.doReturn(room).when(repository).findRoomByRoomId(room.getRoomId());
+        Mockito.doReturn(Arrays.asList(problem, problem2))
+                .when(problemService).getProblemsFromDifficulty(ProblemDifficulty.HARD, 9);
+
+        RoomDto response = gameService.startGame(ROOM_ID, request);
+
+        verify(socketService).sendSocketUpdate(eq(response));
+
+        assertEquals(ROOM_ID, response.getRoomId());
+        assertTrue(response.isActive());
+
+        Game game = gameService.getGameFromRoomId(ROOM_ID);
+        assertNotNull(game);
+
+        // Only 2 problems are selected (no duplicate problems are included)
+        assertEquals(10, game.getRoom().getNumProblems());
+        assertEquals(2, game.getProblems().size());
+        assertEquals(problem.getProblemId(), game.getProblems().get(0).getProblemId());
+        assertEquals(problem.getDifficulty(), game.getProblems().get(0).getDifficulty());
+        assertEquals(problem2.getProblemId(), game.getProblems().get(1).getProblemId());
+        assertEquals(problem2.getDescription(), game.getProblems().get(1).getDescription());
+    }
+
+    @Test
+    public void startGameWithOnlySelectedProblems() {
```

Comment:
Perhaps this test could have two selected problems, as I'm not sure you test that anywhere else?